drivers/wifi/infineon/airoc_wifi.c

```diff
@@ -760,6 +757,80 @@ static int airoc_mgmt_ap_disable(const struct device *dev)
 	return 0;
 }
 
+static int airoc_iface_status(const struct device *dev, struct wifi_iface_status *status)
+{
+	struct airoc_wifi_data *data = dev->data;
+	whd_result_t result;
+	wl_bss_info_t bss_info;
+	whd_security_t security_info = 0;
+	uint32_t wpa_data_rate_value = 0;
+	uint32_t join_status;
+
+	if (airoc_if == NULL) {
+		return -ENOTSUP;
+	}
+
+	status->iface_mode =
+		(data->is_ap_up ? WIFI_MODE_AP
+				: (data->is_sta_connected ? WIFI_MODE_INFRA : WIFI_MODE_UNKNOWN));
+
+	join_status = whd_wifi_is_ready_to_transceive(airoc_if);
+
+	if (join_status == WHD_SUCCESS) {
+		status->state = WIFI_STATE_COMPLETED;
+	} else if (join_status == WHD_JOIN_IN_PROGRESS) {
+		status->state = WIFI_STATE_ASSOCIATING;
+	} else if (join_status == WHD_NOT_KEYED) {
+		status->state = WIFI_STATE_AUTHENTICATING;
+	} else {
+		status->state = WIFI_STATE_DISCONNECTED;
+	}
+
+	result = whd_wifi_get_ap_info(airoc_if, &bss_info, &security_info);
+
+	if (result == WHD_SUCCESS) {
+		memcpy(&(status->bssid[0]), &(bss_info.BSSID), sizeof(whd_mac_t));
+
+		whd_wifi_get_channel(airoc_if, (int *)&status->channel);
+
+		status->band = (status->channel <= CH_MAX_2G_CHANNEL) ? WIFI_FREQ_BAND_2_4_GHZ
```

Comment:
nit, may not be applicable here, but won't be true once 6GHz support is added :).